### PR TITLE
feat: [SRM-13542]: Select monitored services component

### PR DESCRIPTION
**Pull request by karansaraswat19**
#### Screenshots
Screenshot 2023-02-05 at 6 37 22 PM
Screenshot 2023-02-05 at 6 37 35 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by karansaraswat19** [comment]
retrigger jest

**Comment by karansaraswat19** [comment]
retrigger coverage

**Comment by karansaraswat19** [comment]
retrigger jest

**Comment by karansaraswat19** [comment]
retrigger coverage

**Comment by karansaraswat19** [comment]
retrigger jest

**Comment by karansaraswat19** [comment]
retrigger coverage

File: src/modules/85-cv/pages/slos/components/CVCreateDowntime/CVCreateDowntime.tsx [hunk 1]
```diff
@@ -53,21 +83,24 @@ const CVCreateDowntime = (): JSX.Element => {
...
-          runValidationOnMount={Boolean(identifier)}
-          loadingSaveButton={false}
-        />
-      </Formik>
+      <Page.Body loading={loading} error={getErrorMessage(error)} retryOnError={() => refetch()}>
+        {((identifier && data) || !identifier) && (
+          <Formik<DowntimeForm>
+            initialValues={getDowntimeInitialFormData(data?.resource?.downtime)}
+            formName="downtimeForm"
+            onSubmit={values => {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/slos/components/CVCreateDowntime/CVCreateDowntime.tsx:91]:
```suggestion
            onSubmit={handleDowntimeSubmit}
```

File: src/modules/85-cv/pages/slos/components/CVCreateDowntime/components/CreateDowntimeForm/CreateDowntimeForm.tsx [hunk 2]
```diff
@@ -36,57 +39,86 @@ export const CreateDowntimeForm = ({
   )
 
   const [validateAllSteps, setValidateAllSteps] = useState<boolean | undefined>(runValidationOnMount)
+  const [isCreateFlow, setIsCreateFlow] = useState(false)
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/slos/components/CVCreateDowntime/components/CreateDowntimeForm/CreateDowntimeForm.tsx:42]: lets disucss usagae of isCreateFlow

File: src/modules/85-cv/pages/slos/components/CVCreateDowntime/components/CreateDowntimeForm/components/AddMonitoredServices/components/MSList.tsx [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+        />
+      ),
+      id: 'selectMS',
+      width: '50px',
+      Cell: ({ row }: { row: Row<MonitoredServiceListItemDTO> }) => {
+        return <RenderCheckBoxes row={row} selectedMSs={selectedMSs} setSelectedMSs={setSelectedMSs} />
+      }
+    },
+    {
+      accessor: 'name',
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/slos/components/CVCreateDowntime/components/CreateDowntimeForm/components/AddMonitoredServices/components/MSList.tsx:184]: remove accessor , as we don't need sorting

**Review by deepesh-ui — APPROVED** [review]

File: src/modules/85-cv/pages/slos/components/CVCreateDowntime/components/CreateDowntimeForm/CreateDowntimeForm.module.scss [hunk 4]
```diff
@@ -53,3 +53,21 @@
     }
   }
 }
+
+.addMonitoredServices {
+  :global {
+    .bp3-tag {
+      height: fit-content !important;
+      background: #f3f3fa;
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateDowntime/components/CreateDowntimeForm/CreateDowntimeForm.module.scss:61]: use colour from design system

File: src/modules/85-cv/pages/slos/components/CVCreateDowntime/components/CreateDowntimeForm/CreateDowntimeForm.module.scss [hunk 5]
```diff
@@ -53,3 +53,21 @@
...
+      padding: 4px 8px;
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateDowntime/components/CreateDowntimeForm/CreateDowntimeForm.module.scss:62]: use spacing from design system

**Review by praneshg239 — APPROVED** [review]

**Comment by praneshg239** [comment]
@karansaraswat19 please make the commented changes in next PR. Create a ticket for that

**Comment by karansaraswat19** [comment]
retrigger jest

File: src/modules/85-cv/pages/slos/components/CVCreateDowntime/CVCreateDowntime.tsx (hunk 1 above)
Comment (karansaraswat19) [reply src/modules/85-cv/pages/slos/components/CVCreateDowntime/CVCreateDowntime.tsx:91]: bascially handleDowntimeSubmit function type is Promise<void> right now. I am not able to send the function as prop, that's why I have written it as arrow function

**Review by deepesh-ui — APPROVED** [review]

**Review by praneshg239 — APPROVED** [review]

**Pull request merged**